test/jest/batchUploader.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { BatchUploader } from '../../src/batchUploader';
+import { IMParticleWebSDKInstance } from '../../src/mp-instance';
+
+describe('BatchUploader', () => {
+    let batchUploader: BatchUploader;
+    let mockMPInstance: IMParticleWebSDKInstance;
+
+    beforeEach(() => {
+        const now = Date.now();
+        jest.useFakeTimers({
+            now: now,
+            advanceTimers: true // This improves the performance of nested timers, equivalent to Sinon's shouldAdvanceTime
+        });
+        
+        // Create a mock mParticle instance with mocked methods for instantiating a BatchUploader
+        mockMPInstance = {
+            _Store: {
+                SDKConfig: {
+                    flags: {}
+                }
+            },
+            _Helpers: {
+                getFeatureFlag: jest.fn().mockReturnValue(false),
+                createServiceUrl: jest.fn().mockReturnValue('https://mock-url.com')
+            },
+            Identity: {
+                getCurrentUser: jest.fn().mockReturnValue({
+                    getMPID: () => 'test-mpid'
+                })
+            }
+        } as unknown as IMParticleWebSDKInstance;
+
+        batchUploader = new BatchUploader(mockMPInstance, 1000);
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
+    });
+
+    describe('shouldDebounceAST', () => {
+        it('should return false for first call', () => {
+            // First call should not be debounced
+            const firstCall = (batchUploader as any).shouldDebounceAndUpdateLastASTTime();
```

Comment:
Are you setting `batchUploader as any` because the method is private? Does this really need to be private?

If we set this as `any`, I think we lose the benefit of type safety on this object in the tests. I think a cleaner approach would be to the function and use the proper return types.

```
const shouldDebounceAndUpdateLastASTTime: boolean  = (batchUploader as any).shouldDebounceAndUpdateLastASTTime;
```

This would likely clean up your tests and reduce the need to repeat `batchUploader as any`.

If that doesn't work, I would consider creating a `mockBatchUploader`:

```
const mockBatchUploader = batchUploader as any;
``` 

...similar to how we've done workarounds for type hints for invalid values.